src/main/js/ephox/imagetools/transformations/ImageTools.js

```diff
@@ -80,13 +82,13 @@ define("ephox/imagetools/transformations/ImageTools", [
 
   function resize(blob, w, h) {
     return Conversions.blobToImage(blob).then(function(image) {
-      var canvas = Canvas.create(w, h),
-        context = Canvas.get2dContext(canvas);
-
-      context.drawImage(image, 0, 0, w, h);
-      revokeImageUrl(image);
-
-      return Conversions.canvasToBlob(canvas, blob.type);
+      return ImageResizerWebgl.scale(image, w, h)
```

Comment:
There are two parts to this - technology and algorithms. For technology we have two - basic canvas (2d context) and WebGL (also it might spread to Web Workers if possible). And for algorithms - there are quite a lot of them, with different priorities, and most importantly all of them can be implemented using both technologies.

My understanding was, that while it is logical to expose algorithms (and that's how it should be) technology part should be transparent to the end user, since it is obvious that WebGL (if it available) is faster and should be tried in first place. So 2d implementation takes over only if WebGL fails. What might be typical exceptions to this rule?